client/my-sites/checkout/composite-checkout/components/contact-details-container.tsx

```diff
@@ -70,7 +75,14 @@ export default function ContactDetailsContainer( {
 						domainNames={ domainNames }
 						contactDetails={ contactDetails }
 						contactDetailsErrors={ contactDetailsErrors }
-						updateDomainContactFields={ updateDomainContactFields }
+						updateDomainContactFields={ ( details: DomainContactDetailsData ) => {
+							updateDomainContactFields( details );
+							updateRequiredDomainFields( {
+								postalCode:
+									countries?.find( ( country ) => country.code === details.countryCode )
+										?.has_postal_codes ?? true,
+							} );
+						} }
```

Comment:
I'd prefer to define this function outside and just reference it here, instead of defining it in-place, but that's personal preference.